addons/field_rations/XEH_preInit.sqf

```diff
@@ -22,4 +22,20 @@ GVAR(waterSourceOffsets) = [
     [0, 0, 0]
 ];
 
+// Fill water source arrays from CfgVehicles
+{
+    private _split = (getText (_x >> "model")) splitString "\";
+    private _string = toLower (_split select ((count _split) - 1));
+
+    // Append extension if necessary
+    if ((_string select [count _string - 4]) isEqualTo ".p3d") then {
+        _string = _string + ".p3d"
+    };
+
+    private _index = GVAR(waterSourceP3Ds) pushBackUnique _string;
+    if (_index != -1) then {
+        GVAR(waterSourceOffsets) pushBack ([_x >> QGVAR(offset), "ARRAY", [0, 0, 0]] call CBA_fnc_getConfigEntry);
+    };
+} forEach (QUOTE(isNumber (_x >> 'GVAR(waterSupply)')) configClasses (configFile >> "CfgVehicles"));
```

Comment:
Might be a good idea to check that `GVAR(waterSupply)` does not equal `REFILL_WATER_DISABLED`. Isn't the case for acex config but you never know 😃 .